src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java

```diff
@@ -0,0 +1,171 @@
+package core.basesyntax.service;
+/*
+    RULES:
+        Null throws CUSTOM unchecked exception;
+        User's age/login/password can't be null;
+        Age must be in range 18 - 140 inclusively;
+        Login length must be in range 4 - 16 inclusively;
+        Password length must be in range 6 - 16 inclusively;
+        User must have new unique login;
+        Correct users must be added;
+        Login must fit word pattern [a-zA-Z_0-9];
+        Login must not have duplicates regardless to upperCase;
+*/
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+import core.basesyntax.dao.StorageDao;
+import core.basesyntax.dao.StorageDaoImpl;
+import core.basesyntax.db.Storage;
+import core.basesyntax.exceptions.InvalidUserInputException;
+import core.basesyntax.model.User;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+class RegistrationServiceImplTest {
+    private static final String INPUT_EXCEPTION_MESSAGE
+            = "must throw InvalidUserInputException if input is invalid";
+    private static final String USER_RETURN_EXCEPTION_MESSAGE =
+            "Registration should return valid registered user";
+    private static final String DUPLICATED_LOGIN_MESSAGE =
+            "Duplicated login should throw InvalidUserInputException";
```

Comment:
there is no need to make these messages constants
let's maybe write them as text

Reply:
They are being repeated...

Reply:
but they are concatenate, so it is probably better not to make constants out of it - more strings are created